Main.java

```diff
@@ -1,588 +1,287 @@
+        this.itemNo = itemNo;
+        this.quantity = quantity;
+        this.price = calculatePrice(itemNo, quantity);
+    }
+
+    private float calculatePrice(int itemNo, int quantity) {
+        switch (itemNo) {
+            case 1: return quantity * 50;
+            case 2: return quantity * 60;
+            case 3: return quantity * 70;
+            case 4: return quantity * 30;
+            default: return 0;
         }
     }
 }
-class Singleroom implements Serializable
-{
+
+class SingleRoom implements Serializable {
     String name;
     String contact;
     String gender;   
-    ArrayList<Food> food =new ArrayList<>();
+    ArrayList<Food> foodOrders = new ArrayList<>();
 
-   
-    Singleroom()
-    {
-        this.name="";
-    }
-    Singleroom(String name,String contact,String gender)
-    {
-        this.name=name;
-        this.contact=contact;
-        this.gender=gender;
+    SingleRoom(String name, String contact, String gender) {
+        this.name = name;
+        this.contact = contact;
+        this.gender = gender;
     }
 }
-class Doubleroom extends Singleroom implements Serializable
-{ 
+
+class DoubleRoom extends SingleRoom {
     String name2;
     String contact2;
     String gender2;  
-    
-    Doubleroom()
-    {
-        this.name="";
-        this.name2="";
-    }
-    Doubleroom(String name,String contact,String gender,String name2,String contact2,String gender2)
-    {
-        this.name=name;
-        this.contact=contact;
-        this.gender=gender;
-        this.name2=name2;
-        this.contact2=contact2;
-        this.gender2=gender2;
+
+    DoubleRoom(String name, String contact, String gender, String name2, String contact2, String gender2) {
+        super(name, contact, gender);
+        this.name2 = name2;
+        this.contact2 = contact2;
+        this.gender2 = gender2;
     }
 }
-class NotAvailable extends Exception
-{
+
+class NotAvailable extends Exception {
     @Override
-    public String toString()
-    {
-        return "Not Available !";
+    public String toString() {
+        return "Not Available!";
     }
 }
 
-class holder implements Serializable
-{
-    Doubleroom luxury_doublerrom[]=new Doubleroom[10]; //Luxury
-    Doubleroom deluxe_doublerrom[]=new Doubleroom[20]; //Deluxe
-    Singleroom luxury_singleerrom[]=new Singleroom[10]; //Luxury
-    Singleroom deluxe_singleerrom[]=new Singleroom[20]; //Deluxe
+class HotelRooms implements Serializable {
+    DoubleRoom[] luxuryDoubleRooms = new DoubleRoom[10]; 
+    DoubleRoom[] deluxeDoubleRooms = new DoubleRoom[20]; 
+    SingleRoom[] luxurySingleRooms = new SingleRoom[10]; 
+    SingleRoom[] deluxeSingleRooms = new SingleRoom[20]; 
 }
 
-class Hotel
-{
-    static holder hotel_ob=new holder();
-    static Scanner sc = new Scanner(System.in);
-    static void CustDetails(int i,int rn)
-    {
+class Hotel {
+    static HotelRooms hotelData = new HotelRooms();
+    static Scanner scanner = new Scanner(System.in);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/shouryaj98/Hotel-Management-Project-Java/pull/54#issuecomment-2437632510>#b99405</a><span> - 10/25/2024, 12:22 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Creating a new Scanner object for System.in in a static field can lead to resource leaks if not properly closed.
</div>
<div id="fix">
<b>Fix:</b> Instead of creating a static Scanner, consider creating and closing the Scanner within methods that require user input. This ensures proper resource management and prevents potential resource leaks.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -69 +69,2 @@
 -    static Scanner scanner = new Scanner(System.in);
 +    // Remove static Scanner
 +    // Use local Scanner objects in methods that require user input
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged